mockfactory/blinding/catalog.py

```diff
@@ -531,34 +545,54 @@ def Tk(k):
             pphi_prim = 9 / 25 * 2 * np.pi**2 / k**3 * pk_prim(k) / self.cosmo_fid.h**3
             return (pk_lin(k) / pphi_prim)**0.5
 
+        def W(k, sigma=smoothing_radius):
+            return np.exp(- 0.5 * k**2 * sigma**2)
+
         for kslab, slab in zip(mesh.slabs.x, mesh.slabs):
             k = sum(kk.real**2 for kk in kslab)**0.5
             nonzero = k != 0.
-            # minus because applied to randoms
             slab[nonzero] *= bfnl / Tk(k[nonzero])
             slab[~nonzero] = 0.
+
+        if shotnoise_correction:
+            k_pivot, mu_pivot = 8e-3, 0.6  # compute the correctif factor at k_pivot
+
+            shotnoise = 1 / recon._readout(inv_shotnoise, data_positions)
```

Comment:
should be read at randoms_positions in case 'randoms' in method

Reply:
Yes for sure ...